### PR TITLE
[clang][Driver] Add flag for setting SkipFunctionBodies

### PR DESCRIPTION
**Pull request by kadircet**
This is an option set by certain tools (clangd and ASTUnit). Sometimes
there are crashes in clang unique to this configuration and it's really
hard to provide reproducers without invoking the tool.

### CODE REVIEW

File: clang/test/Frontend/skip-function-bodies.cc
Comment (ilya-biryukov) [inline clang/test/Frontend/skip-function-bodies.cc:3]: Could you also check in the test that this flag is not available outside of `-cc1`?

I think we really want to avoid that flag in the user-facing interface as it is expected to break a lot of things.

**Comment by ilya-biryukov** [comment]
I am generally supportive of the idea to expose this for more easily reproducing bugs in source tools, and clangd, in particular.
However, I think it'd be useful to make sure the Clang maintainers are on board.
@cor3ntin, @AaronBallman do you  have any objection to exposing this option?

**Review by ilya-biryukov — COMMENTED** [review]
Let's give other folks a chance to reply before landing this.

**Review by AaronBallman — COMMENTED** [review]
My initial thinking was that this doesn't scale particularly well as there may be other options we want to also set at the same time to help produce a reproducer. But after looking through the existing options, I don't really see others that don't already have some form of option. But I'll still ask the question: would it make sense to name this `-helpful-reproducers` or something more generic so we don't feel tempted to expose even more internal options this way?

That said, so long as this is a `-cc1`-only option, it seems reasonable to me.

**Comment by MaskRay** [comment]
The subject could be clarified: "Add cc1 option for ..." so that readers are clear this is not a driver option.

**Comment by kadircet** [comment]
> That said, so long as this is a -cc1-only option, it seems reasonable to m

this is a cc1-only option. added test, updated the commit message.

File: clang/test/Frontend/skip-function-bodies.cpp
Comment (AaronBallman) [inline clang/test/Frontend/skip-function-bodies.cpp:13]: Yay for a diagnostic, *boo* for it suggesting `-Xclang` in this instance, but I think we can live with it.

**Review by AaronBallman — APPROVED** [review]
LGTM

**Comment by ilya-biryukov** [comment]
> But I'll still ask the question: would it make sense to name this `-helpful-reproducers` or something more generic so we don't feel tempted to expose even more internal options this way?

This is tricky as some issues reproduce only when we skip function bodies and sometimes when we don't. Sometimes we need a PCM that skips function bodies and another file consuming it that does not (this is a normal operation mode of Clangd).
So even this particular flag does not bundle well with any other as we might need to set it independently of everything else.


+1 to Fangrui's suggestion to mention this is a `-cc1` flag in the subject line when committing this.
Otherwise, LGTM.

**Review by ilya-biryukov — APPROVED** [review]

**Pull request merged**
